packages/components/data-table/src/use-manual-column-resizing-reducer.ts

```diff
@@ -1,20 +1,74 @@
-import { useReducer, useDebugValue } from 'react';
+import { useReducer, useDebugValue, MutableRefObject, Dispatch } from 'react';
+import type { TColumn } from './data-table';
+
+type TTableRef = MutableRefObject<HTMLTableElement | undefined>;
+
+type TState = {
+  tableRef?: TTableRef;
+  sizes?: TRenderedColumnMeasurements[];
+  initialColWidth?: number;
+  initialMousePosition?: number;
+  columnBeingResized?: number;
+  hasBeenResized?: boolean;
+};
+
+export type TRenderedColumnMeasurements = {
+  key: string | null;
+  width: number;
+};
+
+type TPayload = {
+  initialColWidth: undefined;
+  initialMousePosition: undefined;
+  columnBeingResized: undefined;
+  hasBeenResized: boolean;
+  sizes: undefined;
+  tableRef: TTableRef;
+};
+
+type TStartResizingPayload = {
+  initialColWidth: number;
+  initialMousePosition: number;
+  columnBeingResized: number;
+} & TState;
+
+type TFinishResizingPayload = {
+  initialColWidth: undefined;
+  initialMousePosition: undefined;
+  columnBeingResized: undefined;
+  hasBeenResized: true;
+} & TState;
+
+type TRegisterColumnMeasurements = {
+  sizes: TRenderedColumnMeasurements[];
+} & TState;
+
+type TAction =
+  | { type: 'reset'; payload?: TPayload }
+  | { type: 'registerColumnMeasurements'; payload: TRegisterColumnMeasurements }
+  | { type: 'startResizing'; payload: TStartResizingPayload }
+  | { type: 'finishResizing'; payload?: TFinishResizingPayload }
```

Comment:
It does not seem to my like the payload is actually used.

```suggestion
  | { type: 'finishResizing'; }
```

Reply:
you're right. This is not used.